openeo_processes_dask/process_implementations/cubes/aggregate.py

```diff
@@ -109,11 +123,132 @@ def aggregate_temporal_period(
+        keep_attrs=True,
+        ignore_nodata=True,
+        positional_parameters=positional_parameters,
+    )
+    result = stat_within_polygon.values
+
+    del masked_data, stat_within_polygon
+    gc.collect()
+
+    return result.T
+
+
 def aggregate_spatial(
     data: RasterCube,
-    geometries: VectorCube,
+    geometries,
     reducer: Callable,
-    target_dimension: str = "result",
-    **kwargs,
+    chunk_size: int = 2,
 ) -> VectorCube:
-    raise NotImplementedError
+    t_dim = data.openeo.temporal_dims
+    t_dim_name = t_dim[0]
+    t_dim = None if len(t_dim) == 0 else t_dim[0]
+    b_dim = data.openeo.band_dims
+    b_dim = None if len(b_dim) == 0 else b_dim[0]
+    
+    if "type" in geometries and geometries["type"] == "FeatureCollection":
+        gdf = gpd.GeoDataFrame.from_features(geometries, DEFAULT_CRS)
+    elif "type" in geometries and geometries["type"] in ["Polygon"]:
+        polygon = shapely.geometry.Polygon(geometries["coordinates"][0])
+        gdf = gpd.GeoDataFrame(geometry=[polygon])
+        gdf.crs = DEFAULT_CRS
+        
+    transform = data.rio.transform()
+    geometries = gdf.geometry.values
+
+    geometry_chunks = [
+        geometries[i : i + chunk_size] for i in range(0, len(geometries), chunk_size)
+    ]
+
+    computed_results = []
+    logger.info(f"Running aggregate_spatial process")
+    try:
+        for i, chunk in enumerate(geometry_chunks):
+            # Create a list of delayed objects for the current chunk
+            chunk_results = Parallel(n_jobs=-1)(
+                delayed(_aggregate_geometry)(
+                    data, geom, transform=transform, reducer=reducer
+                )
+                for geom in chunk
+            )
+            computed_results.extend(chunk_results)
+    except:
+        logger.debug(f"Running process failed at {(i+1) *2} geometry")
+
+    logger.info(f"Finish aggregate_spatial process for {len(geometries)}")
+
+    final_results = np.stack(computed_results)
+    del chunk_results, geometry_chunks, computed_results
+    gc.collect()
+
+    df = pd.DataFrame()
+    keys_items = {}
+
+    for idx, b in enumerate(data[b_dim].values):
+        columns = []
+        for t in range(len(data[t_dim_name])):
+            columns.append(f"{b}_time{t+1}")
+
+        keys_items[b] = columns
+        aggregated_data = final_results[:, idx, :]
+        # Create a new DataFrame with the current data and columns
+        band_df = pd.DataFrame(aggregated_data, columns=columns)
+        # Concatenate the new DataFrame with the existing DataFrame
+        df = pd.concat([df, band_df], axis=1)
+
+    df = gpd.GeoDataFrame(df, geometry=gdf.geometry)
+
+    times = list(data[t_dim_name].values)
+
+    data_vars = {}
+    for key in keys_items.keys():
+        data_vars[key] = (["geometry", t_dim], df[keys_items[key]])
+
+    ## Create VectorCube
+    vec_cube = xr.Dataset(
+        data_vars=data_vars, coords=dict(geometry=df.geometry, time=times)
```

Comment:
Instead of naming the temporal dimension "time" per default, it makes sense to name it after the temporal dimension in the input data, what do you think? 
I think, this could be achieved by something like

`coords= {"geometry": df.geometry, t_dim: times}`

Reply:
Indeed, it makes sense to keep the name as it is. Unfortunately, I currently don't have access to my VM due to internal maintenance. However, you can go ahead and implement the new modifications. 

The name of time dimension has already been extracted in the beginning with `t_dim_name = t_dim[0]`, so we can reuse it here.

Reply:
Ah, I was looking into it, but you were faster with your updates, I now also added a test for a dataset without time.